### PR TITLE
Add FFmpeg and OpenH264 projects

### PR DESCRIPTION
**Pull request by bnhhughes**
Also updates ZooKeeper version from 3.4.6 to 3.4.9 and various other cleanup items.

### CODE REVIEW

**Comment by bnhhughes** [comment]
#approved

File: projects/use/useop-openh264-config.cmake [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+set(OPENH264_INCLUDE_DIR ${OP_ROOTDIR}/include/wels)
+set(OPENH264_LIB_DIR ${OP_ROOTDIR}/lib)
+
+# determine OP_BUILD_STATIC
+include(${CMAKE_CURRENT_LIST_DIR}/opopts.cmake)
+
+string(TOLOWER ${CMAKE_BUILD_TYPE} build_type)
+
+if(OP_BUILD_STATIC)
+  if(WIN32)
```
Comment (swflb) [inline projects/use/useop-openh264-config.cmake:17]: Is there a cmake check for win64?

**Review by swflb — APPROVED** [review]

File: projects/use/useop-openh264-config.cmake (hunk 1 above)
Comment (bnhhughes) [reply projects/use/useop-openh264-config.cmake:17]: Yes - this still holds true for 64-bit builds, this flag is true if the build is on Windows (see https://cmake.org/Wiki/CMake_Useful_Variables and look for WIN32)

**Comment by bnhhughes** [comment]
Any comments that @brobeson or @incessantier can be addressed in the next round of changes, merging this in.

File: projects/ffmpeg.cmake [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+set(FFMPEG_SRC_PATH ${CMAKE_BINARY_DIR}/xpbase/Source/ffmpeg)
+set(FFMPEG_DOWNLOAD_FILE ffmpeg-${VER}.tar.bz2)
+set(PRO_FFMPEG
+  NAME ffmpeg
+  WEB "FFmpeg" https://www.ffmpeg.org/ "FFmpeg website"
+  LICENSE "LGPL" https://www.ffmpeg.org/legal.html "Lesser GPL v2.1"
+  DESC "complete, cross-platform solution to record, convert and stream audio and video"
+  REPO "repo" ${REPO} "forked FFmpeg repo on github"
+  VER ${VER}
+  GIT_ORIGIN git://github.com/ndrasmussen/FFmpeg.git
```
Comment (brobeson) [inline projects/ffmpeg.cmake:15]: I wonder if we should fork FFmpeg (and others) with the distributePro organization. Something to consider for the future.

File: projects/zookeeper.cmake [hunk 3]
```diff
@@ -5,12 +5,12 @@
...
 xpProOption(zookeeper)
-set(ZK_REPO https://github.com/apache/zookeeper.git)
-set(ZK_REPO_PATH ${CMAKE_BINARY_DIR}/xpbase/Source/zookeeper_repo)
+set(ZK_REPO https://github.com/apache/zookeeper)
+set(ZK_REPO_PATH ${CMAKE_BINARY_DIR}/xpbase/Source/zookeeper)
 set(ZK_SRC_PATH ${ZK_REPO_PATH}/src/c/src)
 set(ZK_INCLUDE_PATH ${ZK_REPO_PATH}/src/c/include)
 set(ZK_INSTALL_PATH ${CMAKE_BINARY_DIR}/xpbase/Install/zookeeper)
-set(ZK_VER 3.4.6)
+set(ZK_VER 3.4.9)
```
Comment (brobeson) [inline projects/zookeeper.cmake:13]: Why the version bump? Was this discussed while I was vacation?

File: projects/ffmpeg.cmake (hunk 2 above)
Comment (bnhhughes) [reply projects/ffmpeg.cmake:15]: We have forked a few of the repos with the distributePro organization, good point that we should do the same with FFmpeg, I'll look to do that as part of the cleanup I'm doing.

File: projects/zookeeper.cmake (hunk 3 above)
Comment (bnhhughes) [reply projects/zookeeper.cmake:13]: We're allowed to use 3.4.x so we decided to update to the latest.

**Pull request merged**